pkg/csi/manila/capabilities/manilacapabilities.go

```diff
@@ -33,23 +32,27 @@ const (
 	ManilaCapabilityNone ManilaCapability = iota
 	ManilaCapabilitySnapshot
 	ManilaCapabilityShareFromSnapshot
-)
 
-var (
-	manilaCapabilitiesByShareTypeIDMtx sync.RWMutex
-	manilaCapabilitiesByShareTypeID    = make(map[string]ManilaCapabilities)
+	extraSpecSnapshotSupport                = "snapshot_support"
+	extraSpecCreateShareFromSnapshotSupport = "create_share_from_snapshot_support"
 )
 
 func GetManilaCapabilities(shareType string, manilaClient manilaclient.Interface) (ManilaCapabilities, error) {
-	const (
-		snapshotSupport                = "snapshot_support"
-		createShareFromSnapshotSupport = "create_share_from_snapshot_support"
-	)
+	shareTypes, err := manilaClient.GetShareTypes()
+	if err != nil {
+		return nil, err
+	}
+
+	for _, t := range shareTypes {
+		if t.Name == shareType || t.ID == shareType {
+			return readManilaCaps(t.ExtraSpecs), nil
+		}
+	}
 
-	manilaCapabilitiesByShareTypeIDMtx.RLock()
-	caps, ok := manilaCapabilitiesByShareTypeID[shareType]
-	manilaCapabilitiesByShareTypeIDMtx.RUnlock()
+	return nil, fmt.Errorf("unknown share type %s", shareType)
+}
 
+func readManilaCaps(extraSpecs map[string]interface{}) ManilaCapabilities {
 	strToBool := func(ss interface{}) bool {
```

Comment:
Not for this PR but this could become a more general utility function that could be used e.g. in pkg/ingress/controller/controller.go and pkg/autohealing/cloudprovider/openstack/provider.go as well.